### ISSUE TITLE
Replace pytest-responses with custom code

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-server#10399]
Reading upstream comments about upgrading to pytest 3.6+ in https://github.com/getsentry/pytest-responses/issues/8 makes me think if we should just pull that code into addons-server, it's not that much and should be fairly easy to maintain.

**Comment by eviljeff** [comment]
I agree.  It's looking like that repo is unmaintained, from the lack of attention to the pr/issue (especially vs. the other getsentry repos)
